core/src/main/java/org/bitcoinj/core/ECKey.java

```diff
@@ -186,16 +191,24 @@ protected ECKey(@Nullable BigInteger priv, ECPoint pub) {
      * Utility for compressing an elliptic curve point. Returns the same point if it's already compressed.
      * See the ECKey class docs for a discussion of point compression.
      */
-    public static ECPoint compressPoint(ECPoint uncompressed) {
-        return CURVE.getCurve().decodePoint(uncompressed.getEncoded(true));
+    public static ECPoint compressPoint(ECPoint point) {
+        return point.isCompressed() ? point : CURVE.getCurve().decodePoint(point.getEncoded(true));
```

Comment:
All of these methods are deprecated, and also createPoint() takes BigInteger's. Do you have a non-deprecated proposal?